src/Controllers/AdminController.php

```
@@ -778,6 +801,148 @@ public function billing(): void
+     * @return array{0: \DateTimeImmutable, 1: \DateTimeImmutable, 2: \DateTimeImmutable}
+     */
+    private function currentBillingPeriod(?array $subscription): array
+    {
+        $tz = new \DateTimeZone('UTC');
+        $now = ($this->clock)()->setTimezone($tz);
+        $months = max(1, min(12, (int) ($subscription['billing_period_months'] ?? 1)));
+
+        $startedAt = trim((string) ($subscription['started_at'] ?? ''));
+        try {
+            $anchor = $startedAt !== ''
+                ? new \DateTimeImmutable($startedAt, $tz)
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '854', column '23'). (undefined)

(MissingImport)